lua/entities/sammyservers_textscreen/cl_init.lua

```diff
@@ -60,32 +60,59 @@ function ENT:Draw()
 
 end
 
+-- Return whether the first position is in front of the second with the given direction
+local function isInFront(entPos, plyShootPos, direction)
+    local product = (entPos.x - plyShootPos.x) * direction.x +
+                      (entPos.y - plyShootPos.y) * direction.y +
+                      (entPos.z - plyShootPos.z) * direction.z
+    return (product < 0)
+end
+
+local plyShootPos, ang, pos, camangle, showFront -- Less variables being created each frame
 hook.Add( "PostDrawTranslucentRenderables", "SammyServers3D2DTextScreens", function()
+
+	-- Cache the shoot pos for this frame
+	if IsValid(LocalPlayer()) then
```

Comment:
localplayer can not be NULL at this time

Reply:
Thank you.